source/codegen/templates/service_helpers.mako

```diff
@@ -258,8 +258,12 @@ one_of_case_prefix = f'{namespace_prefix}{function_name}Request::{PascalFieldNam
 %>\
 %     if common_helpers.is_struct(parameter) or underlying_param_type == 'ViBoolean':
       std::vector<${underlying_param_type}> ${parameter_name}(${size}, ${underlying_param_type}());
-%     elif service_helpers.is_string_arg(parameter):
+%     elif common_helpers.is_string_arg(parameter):
```

Comment:
Looks like codegen will remain lot simpler if we declared this additional memory as was being done earlier. We won't need any changes during `create_args` as well (just pass `{parameter_name}.data()` for strings) and during `set_response_values` call `response->set_{parameter_name}_raw({parameter_name})` to set raw array pointer to this additional memory location apart from calling `Copy` function to copy as enums as well.

Reply:
changed this